schema.graphql

```diff
@@ -22,3 +22,47 @@ type Payment @entity {
   feeAmountInCrypto: BigDecimal
   maxRateTimespan: Int
 }
+
+enum EscrowState {
+  inEscrow
+  inEmergency
+  inFrozen
+  closed
+}
+
+type Escrow @entity {
+  id: ID!
+  contractAddress: Bytes!
+  reference: Bytes!
+  creationBlock: Int!
+  creationTimestamp: Int!
+  escrowState: EscrowState!
+  tokenAddress: Bytes!
+  amount: BigDecimal!
+  feeAmount: BigDecimal!
+  feeAddress: Bytes!
+  payer: Bytes!
+  payee: Bytes
+  events: [EscrowEvent!] @derivedFrom(field: "escrow")
```

Comment:
Referencing entities between each other so now we can retrieve Escrow entity with a list of EscrowEvents

Reply:
great!